File: chapter-06/exercise-30-multiplication-table/MultiplicationTable.java
class MultiplicationTable {
  private static final int MAX = 12;
  private static final int COLUMN_WIDTH = 4;
  private static final String FORMAT = "%" + COLUMN_WIDTH + "d";
  
  public static void main(String[] args) {
    // header row
    System.out.printf("%" + COLUMN_WIDTH + "s", "");
    for (var i = 0; i <= MAX; i++) {
      System.out.printf(FORMAT, i);
    }
    System.out.println();

    // rows
    for (var i = 0; i <= MAX; i++) {
      // row header
      System.out.printf(FORMAT, i);

      // row
      for (var j = 0; j <= MAX; j++) {
        System.out.printf(FORMAT, i * j);
      }
      System.out.println();
    }
  }
}

File: exercise-30-multiplication-table/multiplication_table.py
# header row
print(f"{'':>4}", end = " ")
for i in range(13):
    print(f"{i:>4}", end = " ")
print()

# rows
for i in range(13):
    print(f"{i:>4}", end = " ") # row header column
    for j in range(13): # columns
        print(f"{i * j:>4}", end = " ")
    print() # finish row

